qos-monitor/src/main/java/eu/arrowhead/core/qos/service/ping/monitor/impl/DefaultExternalPingMonitor.java

```diff
@@ -0,0 +1,204 @@
+	private static final int ICMP_TTL = 255;
+	private static final int OVERHEAD_MULTIPLIER = 2;
+
+	private final ThreadPoolExecutor threadPool = (ThreadPoolExecutor) Executors.newFixedThreadPool(1);
+
+	@Autowired
+	private QoSMonitorDriver driver;
+
+	@Autowired
+	private PingEventProcessor processor;
+
+	@Resource(name = QosMonitorConstants.EVENT_COLLECTOR)
+	private PingEventCollectorTask eventCollector; 
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_NAME_WD)
+	private String externalPingMonitorName;
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_ADDRESS_WD)
+	private String externalPingMonitorAddress;
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_PORT_WD)
+	private int externalPingMonitorPort;
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_PATH_WD)
+	private String externalPingMonitorPath;
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_SECURE_WD)
+	private boolean pingMonitorSecure;
+
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_AUTHINFO_WD)
+	private String externalPingMonitorAuthInfo;
+
+	private SystemRequestDTO pingMonitorSystem;
+
+	private final Logger logger = LogManager.getLogger(DefaultExternalPingMonitor.class);
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	@Override
+	public List<IcmpPingResponse> ping(final String address) {
+		logger.debug("ping statred...");
+
+		final int timeOut = calculateTimeOut();
+		final UUID measurementProcessId = requestExternalMeasurement(address);
+
+		final long startTime = System.currentTimeMillis();
+		final long meausermentExpiryTime = startTime + timeOut;
+
+		try {
+
+			return processor.processEvents(measurementProcessId, meausermentExpiryTime);
+
+		} catch (final Exception ex) {
+
+			logger.debug(ex.getMessage());
+		}
+
+		return null;
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public void init() {
+		logger.debug("initPingMonitorProvider started...");
+
+		pingMonitorSystem = getPingMonitorSystemRequestDTO();
+
+		threadPool.execute(eventCollector);
+
+		driver.checkPingMonitorProviderEchoUri(createPingMonitorProviderEchoUri());
+		driver.subscribeToExternalPingMonitorEvents(pingMonitorSystem);
+	}
+
+	//=================================================================================================
+	// assistant methods
+
+	//-------------------------------------------------------------------------------------------------
+	private UUID requestExternalMeasurement(final String address) {
+		logger.debug("requestExternalMeasurement started...");
+
+		try {
+
+			final IcmpPingRequestACK acknowledgedMeasurmentRequest = driver.requestExternalPingMonitorService(createPingMonitorProviderUri(), createIcmpPingRequest(address));
```

Comment:
typo in variable name measurment => measurement

Reply:
fixed